### PR TITLE
TE-590 Add the ability to reverse the label position

### PR DESCRIPTION
**Pull request by stephenkr**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-590)

### What **one** thing does this PR do?
Add the ability to render the label on the left side of the label.

### Any other notes
screen shot 2018-07-09 at 12 36 42

### CODE REVIEW

File: src/components/elements/Icon/Readme.md [hunk 1]
```diff
@@ -141,6 +141,14 @@ Library icons
 <Icon color='black' isCircular isColorInverted />
 ```
 
+#### Reversed
+
+```jsx
+<Icon label="Search" name="search" />
+<Divider />
```
[the remark this answers is not in the document]
Comment (stephenkr) [reply src/components/elements/Icon/Readme.md]: Oh true, ok I'll make the change.

File: src/components/elements/Icon/component.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ export const Component = ({
   isCircular,
   isColorInverted,
   isDisabled,
+  isReversed,
```
Comment (marcdavi-es) [inline src/components/elements/Icon/component.js]: I think it's better to be more specific. `isReversed` could be interpreted as the icon itself being reversed?

Maybe `isLabelReversed` or even `isLabelLeft` ?

Comment (stephenkr) [reply src/components/elements/Icon/component.js]: Yep good point, I think `isLabelLeft` makes more sense

File: src/components/elements/Icon/component.js [hunk 3]
```diff
@@ -31,10 +32,11 @@ export const Component = ({
     // compatibility with dynamic components e.g. `ToolTip`
     {...otherProps}
   >
+    {!!label && isReversed === true && <Paragraph>{label}</Paragraph>}
```
Comment (marcdavi-es) [inline src/components/elements/Icon/component.js]: No need for comparison here

`!!label && isReversed && ...`

File: src/components/elements/Icon/component.js [hunk 4]
```diff
@@ -31,10 +32,11 @@ export const Component = ({
...
     <svg viewBox="0 0 24 24">
       <path d={getPath(name, path)} fill="currentColor" />
     </svg>
-    {!!label && <Paragraph>{label}</Paragraph>}
+    {!!label && isReversed === false && <Paragraph>{label}</Paragraph>}
```
Comment (marcdavi-es) [inline src/components/elements/Icon/component.js]: Similarly...

No need for comparison here

`!!label && !isReversed && ...`

File: src/styles/semantic/themes/livingstone/elements/icon.variables [hunk 5]
```diff
@@ -32,8 +32,9 @@
 /* Variations */
 @circularPadding: 0.2em 0;
 
-@iconLabelMarginLeft: @5px;
-@iconLabelMargin: 0 0 0 @iconLabelMarginLeft;
+@iconLabelMarginSpacing: @5px;
+@iconLabelMargin: 0 0 0 @iconLabelMarginSpacing;
+@iconReversedLabelMargin: 0 @iconLabelMarginSpacing 0 0;
```
Comment (marcdavi-es) [inline src/styles/semantic/themes/livingstone/elements/icon.variables]: Nice :)

File: src/components/elements/Icon/component.js (hunk 2 above)
Comment (stephenkr) [reply src/components/elements/Icon/component.js]: Actually i think it has to be `isLabelRight`, making the change now

File: src/components/elements/Icon/component.js (hunk 3 above)
Comment (stephenkr) [reply src/components/elements/Icon/component.js]: I would argue that explicit comparisons are a lot more performant than implicit, but if its the style I'm happy to change it.

https://jsperf.com/implicit-vs-explicit-js-conditionals

Comment (marcdavi-es) [reply src/components/elements/Icon/component.js]: 🤔 Is that true (pun intended) for boolean values?

I take your point re coerced values...

Comment (stephenkr) [reply src/components/elements/Icon/component.js]: Yep true :) I'll make the change, but I'd say V8 is optimized for this style when dealing with booleans.

Also crazy when comparing Safari and Chrome.

Comment (marcdavi-es) [reply src/components/elements/Icon/component.js]: Hmm even for booleans explicit is (marginally) faster... https://jsperf.com/implicit-vs-explicit-js-conditionals-boolean-only

Comment (marcdavi-es) [reply src/components/elements/Icon/component.js]: I'm happy to stick to explicit then :)

Comment (stephenkr) [reply src/components/elements/Icon/component.js]: Oh yeah true, the JS compiler must be aware of the type so it has to do less work.

Comment (stephenkr) [reply src/components/elements/Icon/component.js]: TBH, I think it does matter which approach is taken, the comparison is so close.

I'll stick with the style of the file.

**Pull request merged**